Artifacts/windows-add-local-admin/artifact.ps1

```diff
@@ -0,0 +1,137 @@
+    {
+        throw "This method id not supported on powershell 5 and greater"
+    }
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group");
+    $members = @($group.psbase.Invoke("Members"));
+    $Details = @();
+
+    $members | ForEach-Object {
+        $name = $_.GetType().InvokeMember("Name", 'GetProperty', $null, $_, $null)
+        $path = $_.GetType().InvokeMember("ADsPath", 'GetProperty', $null, $_, $null)
+        $schema = $_.GetType().InvokeMember("Schema", 'GetProperty', $null, $_, $null)
+       
+
+        if ($schema -notlike "WinNT://$name/*")
+        {
+            $Type = "Domain"
+            if ($path -like "*/$env:COMPUTERNAME/*")
+            {
+                $Type = "Local"
+            }
+           
+            $Details += New-Object PSObject -Property @{"Account" = $name; "Type" = $type; }
+        }
+    }
+    Write-Output "Members of $GroupName are:"
+    $Details
+}
+
+##############################
+# Function to find the members of a local machine group, usually administrators
+function Get-LocalGroupMembers ([Parameter(Mandatory = $true)] [string] $GroupName)
+{
+    if ($PSVersionTable.PSVersion.Major -lt 5)
+    {
+        return Get-LocalGroupMembersPs3to5 $GroupName
+    }
+
+    $group = [ADSI]("WinNT://$env:COMPUTERNAME/$GroupName, group");
+    $members = @($group.psbase.Invoke("Members"));
+    $Details = @();
+    
+    $members | ForEach-Object {
+        $name = $_.GetType.Invoke().InvokeMember("Name", 'GetProperty', $null, $_, $null)
+        $path = $_.GetType.Invoke().InvokeMember("ADsPath", 'GetProperty', $null, $_, $null)
+        $schema = $_.GetType.Invoke().InvokeMember("Schema", 'GetProperty', $null, $_, $null)
+
+        if ($schema -notlike "WinNT://$name/*")
+        {
+            $Type = "Domain"
+            if ($path -like "*/$env:COMPUTERNAME/*")
+            {
+                $Type = "Local"
+            }
+           
+            $Details += New-Object PSObject -Property @{"Group" = $GroupName; "Account" = $name; "Type" = $type; }
+        }
+    }
+    Write-Output "Members of $GroupName are:"
+    $Details.ForEach( {$_}) | Format-Table -Property Account, Type -AutoSize
+}
+    
+##############################
+# Function to get the AD User string in format "WinNT://<domain>/<username>"
+function Get-AdUsernameString ([Parameter(Mandatory = $true)] [string] $UserName)
+{
+    if ([string]::IsNullOrEmpty($Username)) 
+    { 
+        Write-Error "Username not provided"
+        throw "Username not provided"
+    }
+
+    if ($Username -notmatch '\\')
+    {
+        $ADResolved = ($Username -split '@')[0]
+        $DomainResolved = ($Username -split '@')[1]    
+        $Username = 'WinNT://', "$DomainResolved", '/', $ADResolved -join ''
+    }
+    else
+    {
+        $ADResolved = ($Username -split '\\')[1]
+        $DomainResolved = ($Username -split '\\')[0]
+        $Username = 'WinNT://', $DomainResolved, '/', $ADResolved -join ''
```

Comment:
Ditto on the syntax. In fact, you can remove this line from both the `if / else` condition and just return it:
`return "WinNT://$DomainResolved/$ADResolved"`. The `if / else` condition is only required to get the correct parts. Once we have them we just need to return the AD user name string.